lfs/client.go

```diff
@@ -570,27 +566,27 @@ func newApiRequest(method, oid string) (*http.Request, Creds, error) {
 		return nil, nil, err
 	}
 
-	req, creds, err := newClientRequest(method, u.String())
+	req, creds, err := newClientRequest(method, u.String(), res.Header)
 	if err != nil {
 		return nil, nil, err
 	}
 
 	req.Header.Set("Accept", mediaType)
-	if res.Header != nil {
-		for key, value := range res.Header {
-			req.Header.Set(key, value)
-		}
-	}
-
 	return req, creds, nil
 }
 
-func newClientRequest(method, rawurl string) (*http.Request, Creds, error) {
+func newClientRequest(method, rawurl string, header map[string]string) (*http.Request, Creds, error) {
 	req, err := http.NewRequest(method, rawurl, nil)
 	if err != nil {
 		return nil, nil, err
 	}
 
+	if header != nil {
```

Comment:
This `nil` check isn't necessary. When passing `nil`, `header` ends up being the zero value for a `map[string]string` which can be ranged over just fine.